geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/SizeableObject2ObjectOpenCustomHashMapWithCursor.java

```diff
@@ -183,6 +192,50 @@ private int hash(K key) {
     return mix(strategy.hashCode(key)) & mask;
   }
 
+  @Override
+  public V put(K k, V v) {
+    V oldValue = super.put(k, v);
+    if (oldValue == null) {
+      // A create
+      arrayContentsOverhead += (int) (elementSizer.sizeof(k) + elementSizer.sizeof(v));
+    } else {
+      // An update
+      arrayContentsOverhead += (int) (elementSizer.sizeof(v) - elementSizer.sizeof(oldValue));
+    }
+    return oldValue;
+  }
+
+  @Override
+  public V remove(Object k) {
+    V oldValue = super.remove(k);
+    if (oldValue != null) {
+      arrayContentsOverhead -= (elementSizer.sizeof(k) + elementSizer.sizeof(oldValue));
+    }
+    return oldValue;
+  }
+
+  @Override
+  public int getSizeInBytes() {
+    return arrayContentsOverhead + calculateBackingArraysOverhead();
+  }
+
+  @VisibleForTesting
+  int calculateBackingArraysOverhead() {
+    // This formula determined experimentally using tests.
+    return BACKING_ARRAY_OVERHEAD_CONSTANT
+        + BACKING_ARRAY_LENGTH_COEFFICIENT * (key.length + value.length);
```

Comment:
you could use getTotalBackingArrayLength() here

Reply:
Done!